tutorials/application_design/c++11/publisher.cxx

```diff
@@ -0,0 +1,133 @@
+//
+// (c) 2024 Copyright, Real-Time Innovations, Inc.  All rights reserved.
+//
+//  RTI grants Licensee a license to use, modify, compile, and create derivative
+//  works of the Software.  Licensee has the right to distribute object form
+//  only for use with RTI products.  The Software is provided "as is", with no
+//  warranty of any type, including any warranty for fitness for any purpose.
+//  RTI is under no obligation to maintain or support the Software.  RTI shall
+//  not be liable for any incidental or consequential damages arising out of the
+//  use or inability to use the software.
+//
+
+#include <iostream>
+#include <sstream>
+#include <thread>
+
+#include <rti/rti.hpp>
+#include "VehicleModeling.hpp"
+#include "common.hpp"
+
+struct PublisherSimulation {
+    explicit PublisherSimulation(
+            dds::pub::DataWriter<VehicleMetrics> metrics_writer,
+            dds::pub::DataWriter<VehicleTransit> transit_writer)
+            : metrics_writer_(metrics_writer),
+              transit_writer_(transit_writer),
+              vehicle_vin_(utils::new_vin()),
+              vehicle_fuel_(100.0),
+              vehicle_route_(utils::new_route()),
+              vehicle_position_(vehicle_route_[0])
+    {
+    }
+
+    bool has_ended() const
+    {
+        return is_out_of_fuel();
+    }
+
+    void run();
+
+    friend std::string utils::to_string<>(PublisherSimulation const &sim);
+
+private:
+    dds::pub::DataWriter<VehicleMetrics> metrics_writer_;
+    dds::pub::DataWriter<VehicleTransit> transit_writer_;
+
+    std::string vehicle_vin_;
+    double vehicle_fuel_;
+    utils::CoordSequence vehicle_route_;
+    Coord vehicle_position_;
+
+    bool is_out_of_fuel() const
+    {
+        return vehicle_fuel_ <= 0.0;
+    }
+
+    bool is_on_standby() const
+    {
+        return vehicle_route_.empty();
+    }
+};
+
+void PublisherSimulation::run()
+{
+    while (!has_ended()) {
+        metrics_writer_.write(VehicleMetrics { vehicle_vin_, vehicle_fuel_ });
```

Comment:
I'm focusing this example on how it looks in the documentation. If we were to change this, I think the documentation would be worse-off because the value is now part of the class. Also, we would need to change it for all APIs.

Reply:
Why would it look worse in the documentation? In `PublisherSimulation::run()` you would just have to replace stuff like

```
vehicle_route_ = utils::new_route();
```

with:

```
vehicle_transit_.route(utils::new_route());
```

which I think is equally readable.

Is there any specific code snippet you find less clear?

(Since this also affects the Python example, I'm OK if we do this outside this PR)